File: leetcode/1/1.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        m = {}
        for idx, i in enumerate(nums):
            if i in m:
                m[i].append(idx)
            else:
                m[i] = [idx]
        for i in nums:
            j = target-i
            if j != i:
                if j in m:
                    return [m[i][0], m[j][0]]
            else:
                if j in m and len(m[j]) >= 2:
                    return m[j][:2]

    def better(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        dic = {}
        for i in range(0, len(nums)):
            if target-nums[i] in dic:
                return [dic[target-nums[i]], i]
            dic[nums[i]] = i

File: leetcode/1295/1.go
package main

func main() {

}

func findNumbers(nums []int) int {
	ret := 0
	for _, num := range nums {
		if x(num)%2 == 0 {
			ret++
		}
	}
	return ret
}

func x(x int) int {
	ret := 0
	for x != 0 {
		x /= 10
		ret++
	}
	return ret
}
